src/analysis/retail/evoker/augmentation/modules/talents/BlisteringScales.tsx

```diff
@@ -0,0 +1,161 @@
+    this.hasRegenerativeChitin = this.selectedCombatant.hasTalent(
+      TALENTS.REGENERATIVE_CHITIN_TALENT,
+    );
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.BLISTERING_SCALES_DAM),
+      this.onHit,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.REACTIVE_HIDE_BUFF),
+      this.onApply,
+    );
+    this.addEventListener(
+      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.REACTIVE_HIDE_BUFF),
+      this.onStackGain,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.REACTIVE_HIDE_BUFF),
+      this.onRemove,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.ERUPTION_TALENT),
+      this.onCast,
+    );
+  }
+
+  onApply() {
+    this.reactiveHideStacks = 1;
+  }
+
+  onRemove() {
+    this.reactiveHideStacks = 0;
+  }
+
+  onStackGain(event: ApplyBuffStackEvent) {
+    this.reactiveHideStacks = event.stack;
+  }
+
+  onHit(event: DamageEvent) {
+    this.reactiveHideDamage += calculateEffectiveDamage(
+      event,
+      this.reactiveHideStacks * REACTIVE_HIDE_MULTIPLIER,
+    );
+    this.blisteringScalesDamage += event.amount + (event.absorbed ?? 0);
+    this.onHitCount += 1;
+    this.totalStacks += this.reactiveHideStacks;
+  }
+
+  onCast(event: CastEvent) {
+    if (!this.hasRegenerativeChitin) {
+      return;
+    }
+    if (chitinBuffStackGained(event)) {
+      this.stacksGained += 1;
+    } else {
+      this.stacksWasted += 1;
+    }
+  }
+
+  statistic() {
+    const averageStacks = this.totalStacks / this.onHitCount;
+    const damageSources = [
+      {
+        color: 'rgb(123,188,93)',
+        label: 'Stacks gained',
+        valueTooltip: formatNumber(this.stacksGained),
+        value: this.stacksGained,
+      },
+      {
+        color: 'rgb(216,59,59)',
+        label: 'Stacks wasted',
+        valueTooltip: formatNumber(this.stacksWasted),
+        value: this.stacksWasted,
+      },
+    ];
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.OPTIONAL(13)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={
+          <>
+            <li>Damage: {formatNumber(this.reactiveHideDamage + this.blisteringScalesDamage)}</li>
+            {this.hasReactiveHide && <li>Average Stacks: {averageStacks.toFixed(2)}</li>}
+          </>
+        }
+      >
+        <TalentSpellText talent={TALENTS.BLISTERING_SCALES_TALENT}>
+          <ItemDamageDone amount={this.blisteringScalesDamage - this.reactiveHideDamage} />
+        </TalentSpellText>
+
+        {this.hasReactiveHide && (
+          <TalentSpellText talent={TALENTS.REACTIVE_HIDE_TALENT}>
+            <ItemDamageDone amount={this.reactiveHideDamage} />
+          </TalentSpellText>
+        )}
+
+        {this.hasRegenerativeChitin && (
+          <div className="pad">
```

Comment:
there's pad here, but not for reactive hide, intentional? (havent checked it out locally)

Reply:
the pad class is needed to make the donut behave properly, and only chitin has a donut